src/go/rdctl/cmd/root.go

```diff
@@ -95,67 +110,75 @@ func getRequestObject(method string, command string) (*http.Request, error) {
   return req, nil
 }
 
-func doRestOfRequest(req *http.Request) error {
-  client := http.Client{}
-  response, err := client.Do(req)
-  if err != nil {
-    return err
-  }
-  if response.StatusCode < 200 || response.StatusCode >= 300 {
-    switch (response.StatusCode) {
-    case 401:
-      return fmt.Errorf("user/password not accepted")
-    case 500:
-      return fmt.Errorf("server-side problem: please consult the server logs for more information.")
-    }
-    return fmt.Errorf("%s", response.Status)
-  }
 
-  defer response.Body.Close()
+func doRestOfRequest(req *http.Request) ([]byte, error) {
+	client := http.Client{}
+	response, err := client.Do(req)
+	if err != nil {
+		return emptyByteArray, err
+	}
+  statusMessage := ""
+	if response.StatusCode < 200 || response.StatusCode >= 300 {
+		switch response.StatusCode {
+		case 400:
+      // rest of message should be in the body
+      statusMessage = "error in request"
+			break
```

Comment:
What is the point of the `break` statement? It is the end of the `case` clause, and they don't fall through to the next case without an explicit `fallthrough`.

Reply:
That code was wrong - the catchall should have been in a `default` handler, not after the switch stmt. Fixed now.